On Ubuntu 24.04, lib_install scripts that rely on checkinstall fail to run.

### Bug

as the title says.
e.g.,
```
junhao@GIGABYTE:~/workspace/clp/components/core/tools/scripts/lib_install$ sudo ./fmtlib.sh 8.0.1
...
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/lib/libfmt.a
CMake Error at cmake_install.cmake:46 (file):
  file INSTALL cannot copy file
  "/tmp/fmtlib-installation/fmt-8.0.1/cmake-build-release/libfmt.a" to
  "/usr/local/lib/libfmt.a": No such file or directory.


make: *** [Makefile:130: install] Error 1

****  Installation failed. Aborting package creation.

Cleaning up...OK

Bye.

```

### CLP version

329edf66e8d368de3070ba266fcf653060fd1541

### Environment

```
junhao@GIGABYTE:~$ uname -a
Linux GIGABYTE 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
junhao@GIGABYTE:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.1 LTS
Release:        24.04
Codename:       noble
junhao@GIGABYTE:~$ cat /etc/os-release
PRETTY_NAME="Ubuntu 24.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.1 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
```

### Reproduction steps

1. `cd ~/workspace/clp/components/core/tools/scripts/lib_install`
2. `sudo ./fmtlib.sh 8.0.1`

--- a/components/core/tools/scripts/lib_install/fmtlib.sh
+++ b/components/core/tools/scripts/lib_install/fmtlib.sh
@@ -38,10 +38,10 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  install_cmd_args+=("sudo")
 fi
 
 # Get number of cpu cores
@@ -74,11 +74,22 @@ checkinstall_installed=$?
 set -e
 
 # Install
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} make install
+install_cmd_args+=(
+  make install
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

--- a/components/core/tools/scripts/lib_install/libarchive.sh
+++ b/components/core/tools/scripts/lib_install/libarchive.sh
@@ -37,10 +37,10 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  install_cmd_args+=("sudo")
 fi
 
 # Get number of cpu cores
@@ -74,11 +74,22 @@ checkinstall_installed=$?
 set -e
 
 # Install
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} make install
+install_cmd_args+=(
+  make install
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

--- a/components/core/tools/scripts/lib_install/liblzma.sh
+++ b/components/core/tools/scripts/lib_install/liblzma.sh
@@ -96,7 +96,7 @@ if [ $checkinstall_installed -eq 0 ] ; then
   )
 fi
 install_cmd_args+=(
-    make install liblzma
+    make install
 )
 "${install_cmd_args[@]}"
 

--- a/components/core/tools/scripts/lib_install/liblzma.sh
+++ b/components/core/tools/scripts/lib_install/liblzma.sh
@@ -37,7 +37,22 @@ if [[ "$#" -gt 1 ]] ; then
   fi
 fi
 
-# Note: we won't check if the package already exists
+# Check if already installed
+set +e
+dpkg -l "${package_name}" | grep "${version}"
+installed=$?
+set -e
+if [ $installed -eq 0 ] ; then
+  # Nothing to do
+  exit
+fi
+
+echo "Checking for elevated privileges..."
+install_cmd_args=()
+if [ ${EUID:-$(id -u)} -ne 0 ] ; then
+  sudo echo "Script can elevate privileges."
+  install_cmd_args+=("sudo")
+fi
 
 # Get number of cpu cores
 num_cpus=$(grep -c ^processor /proc/cpuinfo)
@@ -60,7 +75,30 @@ mkdir build
 cd build
 cmake -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE ../
 make -j${num_cpus}
-make install liblzma
+
+# Check if checkinstall is installed
+set +e
+command -v checkinstall
+checkinstall_installed=$?
+set -e
+
+# Install
+if [ $checkinstall_installed -eq 0 ] ; then
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
+fi
+install_cmd_args+=(
+    make install liblzma
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

--- a/components/core/tools/scripts/lib_install/lz4.sh
+++ b/components/core/tools/scripts/lib_install/lz4.sh
@@ -38,10 +38,10 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  install_cmd_args+=("sudo")
 fi
 
 # Get number of cpu cores
@@ -71,11 +71,22 @@ checkinstall_installed=$?
 set -e
 
 # Install
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} make install
+install_cmd_args+=(
+  make install
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

--- a/components/core/tools/scripts/lib_install/mariadb-connector-c.sh
+++ b/components/core/tools/scripts/lib_install/mariadb-connector-c.sh
@@ -37,10 +37,11 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  privileged_command_prefix="sudo"
+  install_cmd_args+=(${privileged_command_prefix})
 fi
 
 # Get OS version
@@ -77,11 +78,24 @@ set -e
 install_dir=/usr/local
 ${privileged_command_prefix} mkdir -p ${install_dir}
 
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} rsync -a . ${install_dir}/
+install_cmd_args+=(
+  rsync
+  -a .
+  "${install_dir}/"
+)
+"${install_cmd_args[@]}"
 
 # Update ld cache
 ${privileged_command_prefix} ldconfig ${install_dir}/lib/mariadb

--- a/components/core/tools/scripts/lib_install/mongocxx.sh
+++ b/components/core/tools/scripts/lib_install/mongocxx.sh
@@ -85,11 +85,12 @@ set -e
 if [ $checkinstall_installed -eq 0 ] ; then
   install_cmd_args+=(
     checkinstall
+    --default
+    --fstrans=no
+    --nodoc
     --pkgname "${package_name}"
     --pkgversion "${version}"
     --provides "${package_name}"
-    --nodoc
-    -y
     --pakdir "${deb_output_dir}"
   )
 fi

--- a/components/core/tools/scripts/lib_install/msgpack.sh
+++ b/components/core/tools/scripts/lib_install/msgpack.sh
@@ -36,10 +36,10 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  install_cmd_args+=("sudo")
 fi
 
 # Download
@@ -66,11 +66,24 @@ checkinstall_installed=$?
 set -e
 
 # Install
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} cmake --build . --target install
+install_cmd_args+=(
+  cmake
+  --build .
+  --target install
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

--- a/components/core/tools/scripts/lib_install/spdlog.sh
+++ b/components/core/tools/scripts/lib_install/spdlog.sh
@@ -45,10 +45,10 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  install_cmd_args+=("sudo")
 fi
 
 # Get number of cpu cores
@@ -81,12 +81,22 @@ checkinstall_installed=$?
 set -e
 
 # Install
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  # NOTE: We use "1:${version}" to override the version installed by Ubuntu 18.04
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '1:${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} make install
+install_cmd_args+=(
+  make install
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

--- a/components/core/tools/scripts/lib_install/zstandard.sh
+++ b/components/core/tools/scripts/lib_install/zstandard.sh
@@ -38,10 +38,10 @@ if [ $installed -eq 0 ] ; then
 fi
 
 echo "Checking for elevated privileges..."
-privileged_command_prefix=""
+install_cmd_args=()
 if [ ${EUID:-$(id -u)} -ne 0 ] ; then
   sudo echo "Script can elevate privileges."
-  privileged_command_prefix="${privileged_command_prefix} sudo"
+  install_cmd_args+=("sudo")
 fi
 
 # Get number of cpu cores
@@ -74,11 +74,22 @@ checkinstall_installed=$?
 set -e
 
 # Install
-install_command_prefix="${privileged_command_prefix}"
 if [ $checkinstall_installed -eq 0 ] ; then
-  install_command_prefix="${install_command_prefix} checkinstall --pkgname '${package_name}' --pkgversion '${version}' --provides '${package_name}' --nodoc -y --pakdir \"${deb_output_dir}\""
+  install_cmd_args+=(
+    checkinstall
+    --default
+    --fstrans=no
+    --nodoc
+    --pkgname "${package_name}"
+    --pkgversion "${version}"
+    --provides "${package_name}"
+    --pakdir "${deb_output_dir}"
+  )
 fi
-${install_command_prefix} make install
+install_cmd_args+=(
+  make install
+)
+"${install_cmd_args[@]}"
 
 # Clean up
 rm -rf $temp_dir

